Update multiple services example

ker0x opened the pull request:
This PR propose to replace SwiftMailer with the Mailer component and to add the Logger service for a better understanding how to inject multiple services

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Yohann76 committed:
Update messenger.rst

formating improve

lyrixx committed:
Fixed CoverageListener usage with custom SUT solver

See https://github.com/symfony/symfony/blob/d0e5593b1fcc88c0738f1adf2c452f84f19207f3/src/Symfony/Bridge/PhpUnit/Legacy/CoverageListenerTrait.php#L81-L95

ker0x committed:
Update multiple services example

This PR propose to replace SwiftMailer with the Mailer component and to add the Logger service for a better understanding how to inject multiple services

OskarStark commented:
I like the idea but would not add the Logger in this example, yes it makes sense but complicated the example itself

ker0x commented:
Hello @OskarStark , thank you for your feedback!

I have 3 propositions:

1. Removing the Logger from the example. This leave us with a try/catch that only return a boolean, it's a bit strange IMHO
2. Keeping the Logger and update the intro message of the [Handling Multiple Services](https://github.com/symfony/symfony-docs/blob/master/service_container.rst#handling-multiple-services) to specify that we want to log any exception that can be triggered when sending the email.
3. Replacing the Logger with an other component, but which one ?

OskarStark commented:
Please use the same example as before, no need for a try catch here, it’s just an example and should not distract by things which are not needed.

Using

// ... 

Is fine too, meaning there could be some other code parts around. Thanks

HeahDude commented on service_container.rst:
```diff
@@ -311,18 +313,19 @@ made. To do that, you create a new class::
     }
 
-This needs the ``MessageGenerator`` *and* the ``Swift_Mailer`` service. That's no
+This needs the ``MessageGenerator`` *and* the ``Mailer`` service. That's no
```
`MailerInterface`

ker0x replied:
Idk, I'm a little sceptical about this change. We call the `Mailer` service through the `MailerInterface` no?

Would't this sound better ?
> This needs the ``MessageGenerator`` *and* the ``Mailer`` service. That's no
problem, we ask them by type-hinting their class and interface names!

HeahDude replied:
Sounds better indeed

HeahDude reviewed (commented):
Looks good, thanks

javiereguiluz committed:
minor #13251 Fixed CoverageListener usage with custom SUT solver (lyrixx)

This PR was merged into the 3.4 branch.

Discussion
----------

Fixed CoverageListener usage with custom SUT solver

Commits
-------

aeb4053df Fixed CoverageListener usage with custom SUT solver

javiereguiluz committed:
Merge branch '3.4' into 4.4

* 3.4:
  Fixed CoverageListener usage with custom SUT solver

javiereguiluz committed:
minor #13237 Update messenger.rst (Yohann76)

This PR was submitted for the 5.0 branch but it was merged into the 4.4 branch instead (closes #13237).

Discussion
----------

Update messenger.rst

formating improve

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

73e088c78 Update messenger.rst

javiereguiluz commented:
Thanks Romain.
